src/main/java/neo4j/Neo4jDriverAdapter.java

```
@@ -23,33 +29,38 @@ public void close() {
     }
 
     @Override
-    public INode getNode(String key, String value) {
-        StatementResult res = driver.session().beginTransaction()
-                .run("MATCH (n {" + key + ":\"" + value + "\"})RETURN n");
+    public INode getNode(String value) {
+        StatementResult res = driver.session()
+                .run("MATCH (n {name:{value}}) RETURN n",
+                        parameters("value", value));
 
         if (!res.hasNext()) {
             return null;
         }
-        return new Neo4jNode(res.single().get(0).asNode());
+        driver.session().close();
+        Node n = res.single().get("n").asNode();
+        return new Neo4jNode(n);
     }
 
     @Override
     public INode getNode(long id) {
-        String query = "MATCH (n) WHERE id(n)= %s RETURN n";
-        StatementResult res = driver.session().beginTransaction().run(String.format(query, id));
+        StatementResult res = driver.session().run(
+                "MATCH (n) WHERE id(n)={id} RETURN n",
+                parameters("id", id));
 
         if (!res.hasNext()) {
```

Can't you just call next here, then check for a null? That seems cleaner.

---

Calling next wont return null if there are no records, it will throw a NoSuchRecordException. When I was just looking for a Node I did not want to be throwing that, I just wanted to return null if the node isnt found in the db. I could do try/catch, but since we said never to hide exceptions I decided to do this.